velox/dwio/dwrf/test/E2EWriterTest.cpp

```diff
@@ -374,6 +374,93 @@ TEST_F(E2EWriterTest, E2E) {
   dwrf::E2EWriterTestUtil::testWriter(*leafPool_, type, batches, 1, 1, config);
 }
 
+TEST_F(E2EWriterTest, DisableLinearHeuristics) {
+  const size_t batchCount = 100;
+  size_t size = 3000;
+
+  HiveTypeParser parser;
+  auto type = parser.parse(
+      "struct<"
+      "bool_val:boolean,"
+      "byte_val:tinyint,"
+      "short_val:smallint,"
+      "int_val:int,"
+      "long_val:bigint,"
+      "float_val:float,"
+      "double_val:double,"
+      "string_val:string,"
+      "binary_val:binary,"
+      "timestamp_val:timestamp,"
+      "array_val:array<float>,"
+      "map_val:map<int,double>,"
+      "map_val:map<bigint,double>," /* this is column 12 */
+      "map_val:map<bigint,map<string, int>>," /* this is column 13 */
+      "struct_val:struct<a:float,b:double>"
+      ">");
+
+  auto batches = dwrf::E2EWriterTestUtil::generateBatches(
```

Comment:
We can use VectorFuzzer here?

Reply:
Will probably leave that to another diff to fix for all (if it allows seeding). Due to the amount of cpu the writes use, I had to split comparisons in a test into separate test cases, so we want the batches to be generated exactly the same way.